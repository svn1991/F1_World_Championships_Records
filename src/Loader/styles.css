#loading-wrapper {
  width: 100%;
  height: 125px;
  background: white;
  position: relative;
}

#loading-text {
  display: block;
  position: absolute;
  top: 50%;
  left: 50%;
  color: #999;
  width: 100px;
  height: 30px;
  margin: -18px 0 0 -48px;
  text-align: center;
  font-family: "PT Sans Narrow", sans-serif;
  font-size: 10px;
}

#loading-content {
  display: block;
  position: relative;
  width: 100px;
  height: 100px;
  margin: 15px auto;
}

#loading-content:after {
  content: "";
  position: absolute;
  border: 3px solid #0f0;
  left: 15px;
  right: 15px;
  top: 15px;
  bottom: 15px;
}

#loading-content:before {
  content: "";
  position: absolute;
  border: 3px solid #00f;
  left: 5px;
  right: 5px;
  top: 5px;
  bottom: 5px;
}

#loading-content {
  border: 3px solid transparent;
  border-top-color: #4d658d;
  border-bottom-color: #4d658d;
  border-radius: 50%;
  -webkit-animation: loader 2s linear infinite;
  -moz-animation: loader 2s linear infinite;
  -o-animation: loader 2s linear infinite;
  animation: loader 2s linear infinite;
}

#loading-content:before {
  border: 3px solid transparent;
  border-top-color: #d4cc6a;
  border-bottom-color: #d4cc6a;
  border-radius: 50%;
  -webkit-animation: loader 3s linear infinite;
  -moz-animation: loader 2s linear infinite;
  -o-animation: loader 2s linear infinite;
  animation: loader 3s linear infinite;
}

#loading-content:after {
  border: 3px solid transparent;
  border-top-color: #84417c;
  border-bottom-color: #84417c;
  border-radius: 50%;
  -webkit-animation: loader 1.5s linear infinite;
  animation: loader 1.5s linear infinite;
  -moz-animation: loader 2s linear infinite;
  -o-animation: loader 2s linear infinite;
}

@-webkit-keyframes loaders {
  0% {
    -webkit-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }
  100% {
    -webkit-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

@keyframes loader {
  0% {
    -webkit-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }
  100% {
    -webkit-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}
