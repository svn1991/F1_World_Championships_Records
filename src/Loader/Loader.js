import React from 'react';
import './styles.css';

const Loader = () => (
  <div id="loading-wrapper">
    <div id="loading-text">LOADING</div>
    <div id="loading-content" />
  </div>
);

export default Loader;
