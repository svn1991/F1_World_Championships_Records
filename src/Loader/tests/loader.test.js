import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Loader from '../Loader';

describe('<Loader />', () => {
  afterEach(cleanup);

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<Loader />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<Loader />);
    expect(firstChild).toMatchSnapshot();
  });
});
