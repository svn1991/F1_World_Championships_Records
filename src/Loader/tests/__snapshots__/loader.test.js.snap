// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should render and match the snapshot 1`] = `
<div
  id="loading-wrapper"
>
  <div
    id="loading-text"
  >
    LOADING
  </div>
  <div
    id="loading-content"
  />
</div>
`;
